pyteal/ast/abi/tuple.py

```diff
@@ -405,6 +499,13 @@ def store_into(self, output: T) -> Expr:
             output,
         )
 
+    def encode(self) -> Expr:
+        return _index_tuple_bytes(
+            self.tuple.type_spec().value_type_specs(),
+            self.tuple.encode(),
+            self.index,
+        )
```

Comment:
nit: I don't think this should be a special case for only tuple, i.e., this should be a public method in virtual class `ComputedType`, and extend to static/dynamic-array-element. It is non-uniform to let only tuple has such element-wise encoding retrieval.

Reply:
+1 to this. A less-efficient but still correct implementation that could be added to `ComputedValue` could be something like this:

```python
def encode(self) -> Expr:
    return self.use(lambda value: value.encode())
```

And `TupleElement` (and maybe `ArrayElement` eventually) can override this with a more efficient approach, like the one in this PR.

Reply:
I made an attempt to attack in this direction, and a PR up here: https://github.com/algorand/pyteal/pull/629/, please consider a quick pass

Reply:
resolving for #629 is now merged in